src/pages/api/chainlink/productRegistoration.ts

```diff
@@ -1,44 +1,134 @@
 // Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import { Product } from "@shopify/shopify-api/rest/admin/2023-04/product";
 import type { NextApiRequest, NextApiResponse } from "next";
-import { ProductRegistrationReqBody } from "../../../types/ShopifyAPITypes";
+
+import { prisma } from "../../../../lib/prisma";
 
 export default async function handler(
-  req: NextApiRequest & { body: ProductRegistrationReqBody },
+  req: NextApiRequest & {
+    body: {
+      accessToken: string;
+      storeUrl: string;
+      productProfitRightNFTAddress: string;
+      product: Product;
+      price: number;
+      cost: number;
+    };
+  },
   res: NextApiResponse
 ) {
-  console.log(`Chainlink: Product Registration API`);
-  console.log(`method: ${req.method}`);
-  console.log("body: ", req.body);
-  if (req.method !== "POST") {
-    res.statusCode = 405;
-    res.setHeader("Allow", "POST");
-    res.end("Method Not Allowed");
-    return;
-  }
-
-  const {
-    productProfitRightNFTAddress,
-    productJsonString,
-    productVariantsJsonString,
-    productOptionsJsonString,
-    productImagesJsonString,
-    productInitialInventories,
-  } = req.body;
-
-  // Log the received data for debug purposes
-  console.log(
-    "Product Profit Right NFT Address: ",
-    productProfitRightNFTAddress
-  );
-  console.log("Product JSON: ", productJsonString);
-  console.log("Product Variants: ", productVariantsJsonString);
-  console.log("Product Options: ", productOptionsJsonString);
-  console.log("Product Images: ", productImagesJsonString);
-  console.log("Product Initial Inventories: ", productInitialInventories);
-
-  const sampleResponse = {
-    message: "Product Registration Received",
+  const headers = {
+    // Ex shpat_1234567890abcdef1234567890abcdef
+    "X-Shopify-Access-Token": req.body.accessToken,
+    "Content-Type": "application/json",
   };
+  try {
+    console.log(`Chainlink: Product Registration API`);
+    console.log(`method: ${req.method}`);
+    console.log("body: ", req.body);
+    if (req.method !== "POST") {
+      res.statusCode = 405;
+      res.setHeader("Allow", "POST");
+      res.end("Method Not Allowed");
+      return;
+    }
+
+    console.log(`accessToken: ${req.body.accessToken}`);
+    console.log(`storeUrl: ${req.body.storeUrl}`);
+    // Ex https://your-development-store.myshopify.com/admin/api/2023-04/shop.json
+    const productCreateUrl = `https://${req.body.storeUrl}.myshopify.com/admin/api/2023-04/products.json`;
```

Comment:
@sa4mi 
The shopify user can also set up their own domain, would you consider changing it and checking this out?
https://www.shopify.com/jp/blog/setting-up-your-domain

Reply:
@0xShin0221 
Thanks for your feedback!
According to [Shopify Community](https://community.shopify.com/c/shopify-apis-and-sdks/error-using-api-with-custom-domain/td-p/1911795), it seems that using a custom domain for the destination of API requests is not possible. Therefore, it appears that requests need to be sent to a URL with myshopify.com.